templates/dotnet/src/Appwrite/Client.cs.twig

```diff
@@ -172,10 +158,24 @@ namespace {{ spec.title | caseUcfirst }}
                     request.Headers.Add(header.Key, header.Value);
                 }
             }
-            HttpResponseMessage httpResponseMessage = await http.SendAsync(request);
+            try
+            {
+                var httpResponseMessage = await http.SendAsync(request);
+                var code = (int) httpResponseMessage.StatusCode;
+                var response = JObject.Parse(await httpResponseMessage.Content.ReadAsStringAsync());
+                
+                if (code >= 400) {
+                    var message = response["message"].ToString();
+                    throw new AppwriteException(message, code, response.ToString());
+                }
+
+                return httpResponseMessage;
+            }
+            catch (System.Exception e)
+            {
+                throw new AppwriteException(e.Message, e);
```

Comment:
passing second param as exception itself, will it work?

Reply:
Yes, that way I can pass the exception as an inner exception 🙂 

See the 2 constructors here:
https://github.com/appwrite/sdk-generator/blob/feat-dotnet-exceptions/templates/dotnet/src/Appwrite/Models/Exception.cs.twig#L15